main.cpp

```diff
@@ -79,9 +79,9 @@ class Car
 {
 public:
     
-    Car(std::string name_, double speed_, unsigned long passengers_) : name(name_), speed(speed_), passengers(passengers_)
+    Car(std::string name_, double speed_) : name(name_), speed(speed_) 
     {
-        for(size_t i = 0; i < passengers; ++i)
+        for(size_t i = 0; i < passengerRiskArray.size(); ++i)
```

Comment:
`passengerRiskArray.size()` is 0 when this is run.  You still need some initial quantity greater than 0 in order to fill it.
You can keep the constructor parameter `passengers_` and use it here to fill your vector.  Just don't store it as a member variable or use that member variable to keep track of the size of your vector.
- [ ] revise